lib/improver/nowcasting/optical_flow.py

```diff
@@ -321,23 +321,29 @@ def grid_spacing(coord):
         original_datetime, = \
             (cube.coord("time").units).num2date(cube.coord("time").points)
         new_datetime = original_datetime + timestep
-        new_time = (cube.coord("time").units).date2num(new_datetime)
 
-        if np.int64(new_time) == new_time:
-            new_time = np.int64(new_time)
-        else:
-            new_time = np.float64(new_time)
+        new_time = (cube.coord("time").units).date2num(new_datetime)
 
         advected_cube.coord("time").points = new_time
+        advected_cube.coord("time").convert_units(
+            "seconds since 1970-01-01 00:00:00")
+        advected_cube.coord("time").points = (
+            np.around(advected_cube.coord("time").points).astype(np.int64))
+
+        frt_coord = cube.coord("forecast_reference_time")
+        advected_cube.coord("forecast_reference_time").points = (
+            frt_coord.points)
```

Comment:
I think we need to be able to handle the case where the original cube does not have a forecast reference time.  This means the above two lines (which have no effect if a "forecast_reference_time" coord exists, and will throw an error if it doesn't) need expanding I think to:
```
try:
    advected_cube.coord("forecast_reference_time").convert_units(
        "seconds since 1970-01-01 00:00:00")
except CoordinateNotFoundError:
    frt_coord = cube.coord("time").copy()
    frt_coord.rename("forecast_reference_time")
    advected_cube.add_aux_coord(frt_coord)
    advected_cube.coord("forecast_reference_time").convert_units(
        "seconds since 1970-01-01 00:00:00")
```

Reply:
Amended to use a try/except statement for the `forecast_reference_time` coordinate.